src/Forms/GridField/GridFieldDetailForm_ItemRequest.php

```diff
@@ -290,6 +294,45 @@ protected function getFormActions()
                     ->setUseButtonTag(true)
                     ->addExtraClass('btn-outline-danger btn-hide-outline font-icon-trash-bin action--delete'));
             }
+
+            $gridStateStr = $this->getRequest()->requestVar('gridState');
+
+            $this->gridField->getState(false)->setValue($gridStateStr);
+            $actions->push(HiddenField::create('gridState', null, $gridStateStr));
+
+            $rightGroup = CompositeField::create()->setName('RightGroup');
+            $rightGroup->addExtraClass('right');
+            $rightGroup->setFieldHolderTemplate(get_class($rightGroup) . '_holder_buttongroup');
+
+            $previousAndNextGroup = CompositeField::create()->setName('PreviousAndNextGroup');
+            $previousAndNextGroup->addExtraClass('rounded');
+            $previousAndNextGroup->setFieldHolderTemplate(get_class($previousAndNextGroup) . '_holder_buttongroup');
+
+
+            $component = $this->gridField->getConfig()->getComponentByType(GridFieldDetailForm::class);
+
+            if ($component->getShowPagination()) {
+                $previousAndNextGroup->push(FormAction::create('doPrevious')
+                    ->setUseButtonTag(true)
+                    ->setAttribute('data-grid-state', $gridStateStr)
+                    ->setDisabled(!$this->getPreviousRecordID())
+                    ->addExtraClass('btn btn-secondary font-icon-left-open action--previous discard-confirmation'));
+
+                $previousAndNextGroup->push(FormAction::create('doNext')
+                    ->setUseButtonTag(true)
+                    ->setAttribute('data-grid-state', $gridStateStr)
+                    ->setDisabled(!$this->getNextRecordID())
+                    ->addExtraClass('btn btn-secondary font-icon-right-open action--next discard-confirmation'));
+            }
+
+            $rightGroup->push($previousAndNextGroup);
+
+            if ($component->getShowAdd()) {
+                $rightGroup->push(FormAction::create('doNew')
+                    ->setUseButtonTag(true)
+                    ->setAttribute('data-grid-state', $this->getRequest()->getVar('gridState'))
+                    ->addExtraClass('btn btn-primary font-icon-plus rounded action--new discard-confirmation'));
+            }
```

Comment:
There's a lot of code here, are you able to break it up a bit into individual methods?